### ISSUE TITLE
Add support for UTF-8 (Unicode)?

### ISSUE THREAD

**Issue by agarny** [cellml/libcellml#384]
I was just trying to use some Unicode character (e.g. `Λ`) in a generator profile and the generated code was obviously wrong (the Unicode character got converted to `\xCE\x9B`).

I haven't checked in details whether it would be "safe" to use UTF-8, but we probably ought to look into it (FWIW, the CellML API makes extensive use of `std::wstring`).

**Comment by hsorby** [comment]
I don't see any need for UTF-8.  No CellML identifier can use those characters and as far as I know no language requires UTF-8.

**Comment by agarny** [comment]
Indeed, CellML identifiers require Basic Latin characters. However, variable names (in the generated code) might use UTF-8 (e.g. it's allowed in Python 3). The origin comment might also contain some UTF-8 characters. Not everybody uses "simple" characters as native English speakers do... :)

**Comment by hsorby** [comment]
Variable names have to be Basic Latin characters as well.  I don't think we can get UTF-8 characters into a valid CellML model.

**Comment by agarny** [comment]
I was talking about the variables names in the generated code, so things like `VOI_INFO` (which gets set using `GeneratorProfile::setInterfaceVoiInfoString()`). We might, for example, name it `統合変数_情報`.

A more realistic case could be the origin comment string, which by default is set to `The content of this file was generated using <PROFILE_INFORMATION> libCellML <LIBCELLML_VERSION>.` while we might want it to be set to `このファイルのコンテンツは、<PROFILE_INFORMATION> libCellML <LIBCELLML_VERSION>を使用して生成されました。`

Anyway, this is clearly not urgent, but I still think it would be good to look into it at some point.
